.github/workflows/go.yml

```diff
@@ -15,11 +15,37 @@ on:
       - '*.md'
 
 jobs:
+  cover:
+    name: Coverage
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup go
+        uses: actions/setup-go@v5
+        with:
+          cache: true
+          go-version: '1.22'
+
+      - name: Check out code
+        uses: actions/checkout@v4
+
+      - name: Run go test
+        run: go test -coverprofile=coverage.txt -covermode=atomic ./...
+
+      - name: Codecov
+        uses: codecov/codecov-action@v4
+        with:
+          fail_ci_if_error: true # if something is wrong on uploading codecov results, then this job will fail
+          files: ./coverage.txt
+          slug: nspcc-dev/neofs-node
+          token: ${{ secrets.CODECOV_TOKEN }}
+          verbose: true
+
   test:
     runs-on: ubuntu-latest
+    name: Unit tests
     strategy:
       matrix:
-        go: [ '1.20.x', '1.21.x', '1.22' ]
+        go: [ '1.20.x', '1.21.x' ]
```

Comment:
does it solve something? i do not like this version split so would do only if it really helps us with something

Reply:
and is it a problem? we can run tests for 3 versions, and upload (and run test the second time) with the latest one. i do not like the idea of moving the whole version run to a separate job with `Coverage` name when the only reason is "maybe action is a bit more reliable"

Reply:
Likely it can be reworked this way, but an additional job is rather easy to have. It also makes it easier to separate coverage from other tests for various purposes (like branch protection rules).